### PR TITLE
WOC: Implement utm* params of a login UserAction stored in the DB feature #2212

### PR DESCRIPTION
**Pull request by unclebean**
##### Description

This PR is for saving utm* params to DB for data analysis, relevant story is [2212](https://github.com/gitcoinco/web/issues/2212). 
Whenever user access gitcoin with "utm_medum, utm_source, utm_campaign" will store them in cookie, after user login or user logout will store to dashboard_useraction table.

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

UserAction data model.

##### Testing

Test with below 3 scenarios.

**Scenario:** Record utm* in Cookie
**Give:** utm_medum, utm_source, utm_campaign in URL
**When:** User open gitcoin
**Then:** Should record utm* in Cookie

**Scenario:** Record utm* in DB
**Give:** utm* in Cookie
**When:** User click "LOG IN" button
**Then:** Should record utm* in DB

**Scenario:** Record utm* in DB
**Give:** utm* in Cookie
**When:** User click "Log Out" button
**Then:** Should record utm* in DB

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@unclebean could you chery-pick you commits and ensure only those are there in this PR ?

**Comment by unclebean** [comment]
> @unclebean could you chery-pick you commits and ensure only those are there in this PR ?

hi @thelostone-mc actually all commits are related this PR, sorry so many times commit for code format issue.

**Comment by mbeacom** [comment]
@unclebean Hey! Thanks for the contribution!

@thelostone-mc is referring to the bad rebase/merge update that caused many additional changes to be reflected in your `changed files`.  It appears the commits are replaying commits from yesterday and today.  Can you please recreate your commit history here to resolve some of the commit history reflecting unassociated changes?

File: app/dashboard/models.py [hunk 1]
```diff
@@ -2324,6 +2324,9 @@ class UserAction(SuperModel):
     ip_address = models.GenericIPAddressField(null=True)
     location_data = JSONField(default=dict)
     metadata = JSONField(default=dict)
+    utm_campaign = models.CharField(max_length=100, null=True)
```
Comment (mbeacom) [inline app/dashboard/models.py]: Can you adjust this to simply store `utm` data in a `JSONField`?

Comment (unclebean) [reply app/dashboard/models.py]: @mbeacom I have changed utm_campaign to JSON Field, please have a look.

File: pydocmd.yml [hunk 2]
```diff
@@ -130,6 +128,8 @@ generate:
     - gitcoinbot.models++
   - gitcoinbot/views.md:
     - gitcoinbot.views++
+  - healthcheck/healthchecks.md:
+    - healthcheck.healthchecks++
```
[the remark this answers is not in the document]
Comment (unclebean) [reply pydocmd.yml]: hi @mbeacom, seems this stickler-ci issue is your changing. Could you fix it or tell me what's wrong on it?

[the remark this answers is not in the document]
Comment (mbeacom) [reply pydocmd.yml]: @unclebean Your rebase is bad right now.  A lot of commits are being replayed that have already occurred against master.  You'll want to cherry pick your commits to only include your unique changes in this PR.  See:  https://github.com/gitcoinco/web/pull/2309#issuecomment-426090242

[the remark this answers is not in the document]
Comment (unclebean) [reply pydocmd.yml]: hi @mbeacom sorry about that. I should not merge code when I do this PR. I just rebase to one commit only. If I did something wrong please let me know and tell how to resolve it. I want to pick up more :)

**Pull request closed without merging**